### DISCUSSION TITLE
How does penrose compare to Metapost or Tikz?

### DISCUSSION THREAD

**Discussion by ramkumarkb** [penrose/penrose#719] [General]
I have not gone thru the full paper of penrose but a cursory look at the syntax reminded me of Metapost - https://www.tug.org/docs/metapost/mpman.pdf 

So am curious whether any work has been done to compare penrose to similar systems like Metapost or [Tikz](https://tikz.net/) drawing systems.

**Comment by keenancrane** [comment]
The whole point of Penrose is to be one layer of abstraction *above* things like Metapost, TikZ, PostScript, SVG, etc.  The aforementioned are all low-level languages for specifying exactly where and how graphical elements appear on a canvas.  Penrose enables users to instead specify high-level relationships between conceptual objects in a diagram—and then automatically generates output in a low-level language (currently SVG, though nothing about the core design would prevent it from outputting to Metapost, TikZ, PostScript, etc.).

For further discussion, see Section 2.4 of the paper ("Related Systems").
